src/api/client/Network.ts

```diff
@@ -579,4 +579,27 @@ export class Network {
     };
     /* eslint-enable @typescript-eslint/no-non-null-assertion */
   }
+
+  /**
+   * Check if the middleware is up to date with the chain.
+   *
+   * @note Since the middleware can take some time in processing large blocks,
+   * the function returns true if the number of processed blocks by the middleware doesn't lag the finalized chain blocks by more than 20.
+   * More details about the last processed block from middleware can be fetched using the `getMiddlewareMetadata` method.
+   *
+   * @note uses the middleware V2
+   */
+  public async isMiddlewareSynced(): Promise<boolean> {
+    let lastProcessedBlockFromMiddleware = new BigNumber(0);
+    const [latestBlockFromChain, middlewareMetadata] = await Promise.all([
+      this.context.getLatestBlock(),
+      this.getMiddlewareMetadata(),
+    ]);
+
+    if (middlewareMetadata) {
+      lastProcessedBlockFromMiddleware = middlewareMetadata.lastProcessedHeight;
+    }
+
+    return latestBlockFromChain.minus(lastProcessedBlockFromMiddleware).isLessThanOrEqualTo(20);
```

Comment:
Could you extract the number to a const? Also I think 20 is too big. Have you checked in practice how many blocks away is it?

Reply:
If you have seen it on 10, 5 doesn't look safe. In my mind 10 would be enough but if you saw it at that value, maybe 12 or 15 is safer.

Reply:
I think we should return how many blocks its lagging and let the consumer know about it. Seems like an impossible task to find a single number that is OK for every app imo

Comments should mention expected block time is 6 seconds as well